mkdocs/structure/files.py

```diff
@@ -0,0 +1,128 @@
+import fnmatch
+import os
+
+
+class Files(object):
+    def __init__(self, files):
+        self._files = files
+        self.input_paths = {file.input_path: file for file in files}
+
+    def __iter__(self):
+        return iter(self._files)
+
+    def __len__(self):
+        return len(self._files)
+
+    def documentation_pages(self):
+        return [file for file in self if file.is_documentation_page()]
+
+    def static_pages(self):
+        return [file for file in self if file.is_static_page()]
+
+    def media_files(self):
+        return [file for file in self if file.is_media_file()]
+
+    def javascript_files(self):
+        return [file for file in self if file.is_javascript()]
+
+    def css_files(self):
+        return [file for file in self if file.is_css()]
+
+
+class File(object):
+    def __init__(self, from_dir, path, filename=None):
+        # The filename may already been parsed out of the path,
+        # but if not then determine it now.
+        if filename is None:
+            filename = os.path.basename(path)
+
+        root, extension = os.path.splitext(filename)
+        self.root = root  # The filename, without any extension.
+        self.extension = extension.lower()
+
+        self.input_path = path
+        self.full_input_path = os.path.normpath(os.path.join(from_dir, self.input_path))
+        self.output_path = self.get_output_path()
+
+    def get_output_path(self):
+        """
+        Map input source documents to the output build file.
+        """
+        if self.is_documentation_page():
+            if self.root in ('index', 'README'):
+                directory = os.path.dirname(self.input_path)
+            else:
+                directory = os.path.splitext(self.input_path)[0]
+            return os.path.join(directory, 'index.html')
+        return self.input_path
+
+    def is_documentation_page(self):
+        return self.extension in (
+            '.markdown',
```

Comment:
This is duplicated in the urls.py, would it be better to expand/change the is_markdown utility method and use that?